go/vt/vtgate/vtgate.go

```diff
@@ -267,6 +272,9 @@ func (vtg *VTGate) Execute(ctx context.Context, session *vtgatepb.Session, sql s
 	qr, err = vtg.executor.Execute(ctx, "Execute", NewSafeSession(session), sql, bindVariables)
 	if err == nil {
 		vtg.rowsReturned.Add(statsKey, int64(len(qr.Rows)))
+		if len(qr.Rows) == 0 { // HACK to identify writes
```

Comment:
do we have to check that qr.Rows is empty first? couldn't we always write this? most of the time it's going to be zero, right?

Reply:
fix it :100: